packages/@glimmer/runtime/lib/vm/rehydrate-builder.ts

```diff
@@ -45,17 +46,38 @@ export class RehydrateBuilder extends NewElementBuilder implements ElementBuilde
     let node = this.currentCursor!.element.firstChild;
 
     while (node !== null) {
-      if (isComment(node) && isSerializationFirstNode(node)) {
+      if (isOpenBlock(node)) {
         break;
       }
       node = node.nextSibling;
     }
 
-    assert(
-      node,
-      `Must have opening comment <!--${SERIALIZATION_FIRST_NODE_STRING}--> for rehydration.`
-    );
+    assert(node, 'Must have opening comment for rehydration.');
     this.candidate = node;
+    const startingBlockOffset = getBlockDepth(node as SimpleComment);
+    if (startingBlockOffset !== 0) {
+      // We are rehydrating from a partial tree and not the root component
+      // We need to add an extra block before the first block to rehydrate correctly
```

Comment:
Just curious, why do we have to add an extra block for this case?

Reply:
I may be completely off here, but if I remember correctly when we are invoking from `renderComponent` we actually [synthesis an invocation](https://github.com/glimmerjs/glimmer-vm/blob/5ec6a513d7c3ccba9c6b3e572a25162ea5a0a561/packages/%40glimmer/runtime/lib/render.ts#L55). This is going to push blocks onto the stack.

I'm not sure if the test suite is using this under the hood though.

Reply:
That is precisely it, using renderComponent pushes an extra block onto the stack and we use renderComponent in both glimmerx and glimmer.js. The test suite uses renderMain for all the tests. There was one suite that used renderComponent but that suite doesn't seem to be getting executed anymore.

Reply:
Gotcha, thank you for explaining!

Reply:
We should work toward removing `renderMain` in general and refactoring this I think, but that's outside fo the scope for this PR